safedelete/queryset.py

```diff
@@ -107,14 +110,16 @@ def _filter_visibility(self):
                 )
             )
 
+            self._safedelete_filter_applied = True
```

Comment:
Had to do this as _fetch_all is getting used a lot, even after us has retrieved from the db. We can maybe check result_cache of the queryset object instead..